doc/rst/language/spec/conversions.rst

```diff
@@ -218,54 +175,235 @@ type ``C`` is allowed when ``D`` is a subclass of ``C``.
+between two types, the compiler will check that a corresponding ``init=``
+also exists and emit an error if not.  Additionally, if ``init=`` is
+provided to initialize one type from another, the compiler will check
+that a corresponding ``:`` overload exists and will emit an error if not.
+See also :ref:`Explicit_Conversions` for more information on the ``:``
+operator. It is possible to provide ``:`` without ``init=`` or to provide
+``init=`` without ``=``.
+
+   *Example (implementing-assignment.chpl)*
+
+   Suppose that we have defined a record type to wrap an integer:
+
+   .. code-block:: chapel
+
+      record myInteger {
+        var intValue: int;
+      }
+
+   We might wish to support assignments setting a ``myInteger`` from
+   ``int``. In that event, we can provide the following functions:
+
+   .. code-block:: chapel
+
+      operator =(ref lhs: myInteger, rhs: int) {
+        lhs.intValue = rhs;
+      }
+      proc myInteger.init=(rhs: int) {
+        this.intValue = rhs;
+      }
+      operator :(from: int, type toType: myInteger) {
+        var tmp: myInteger = from; // invoke the init= above
+        return tmp;
+      }
+
+   All three of these functions are required if we wish to support
+   assignment. We can invoke these functions like this:
+
+   .. code-block:: chapel
 
-Implicit Conversions To Generic Types
+      var a = 1:myInteger;  // cast -- invokes operator :
+
+      var b: myInteger = 2; // initialization -- invokes init=
+
+      var c: myInteger;
+      c = 3;                // split-initialization -- invokes init=
+
+      var d = new myInteger();
+      d = 4;                // assignment -- invokes operator =
+
+   .. BLOCK-test-chapelnoprint
+
+      writeln("a is ", a, " : ", a.type:string);
+      writeln("b is ", b, " : ", b.type:string);
+      writeln("c is ", c, " : ", c.type:string);
+      writeln("d is ", d, " : ", d.type:string);
+
+   .. BLOCK-test-chapeloutput
+
+      a is (intValue = 1) : myInteger
+      b is (intValue = 2) : myInteger
+      c is (intValue = 3) : myInteger
+      d is (intValue = 4) : myInteger
+
+.. _Implicit_Conversion_Call:
+
+Implicit Conversions for Function Calls
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+An implicit conversion for a function call - also called a *coercion* -
+occurs when the actual argument of a function call is converted to the
+type of the corresponding formal argument, if the formal’s intent is
+``param``, ``in``, ``const in``, or an abstract intent
+(:ref:`Abstract_Intents`) with the semantics of ``in`` or ``const in``.
+These coercions are available among built-in types as described in
+:ref:`Implicit_Conversions`.
+
+Additionally, an implicit conversion for a function call occurs when the
+actual type is a subtype of the formal type. This rule applies to ``in``,
+``const in``, ``const ref``, and ``type`` intent formals and includes
+generic formal types. See :ref:`Subtype_Arg_Conversions`.
```

Comment:
Need to reconcile when subtype conversions occur. We say they occur for fn calls, along with =/init=, at the start of the Implicit Conversions section. They we say some more about fn calls in the Subtype Conversions subsections. They we come back to it again here. If the Subtype Conversions section can define the rules completely, the above paragraph and the following paragraph (about `ref`) would be duplicates and should be deleted. The const-ref open issue below would be moved as well.

Reply:
I opted to make the Subtype Conversions section talk about what are allowed conversions so this section now is the right place to talk about actual and formal types.